edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleIdentityLifecycleManager.cs

```diff
@@ -28,6 +28,18 @@ public ModuleIdentityLifecycleManager(IIdentityManager identityManager, ModuleId
 
         public async Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesAsync(ModuleSet desired, ModuleSet current)
         {
+            IImmutableDictionary<string, Identity> identities = (await this.identityManager.GetIdentities()).ToImmutableDictionary(i => i.ModuleId);
+
+            // Need to remove any identities (except EA/EH) that are managed by EA but don't have a tracked module for in the ModuleSet.
+            IEnumerable<string> removeCurrentIdentities = identities.Where(
+                i => !(Constants.EdgeAgentModuleIdentityName.Equals(i.Key, StringComparison.Ordinal) || Constants.EdgeHubModuleIdentityName.Equals(i.Key, StringComparison.Ordinal))).Where(
+                i => Constants.ModuleIdentityEdgeManagedByValue.Equals(i.Value.ManagedBy, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
are 2 where clauses better here? was that for perf?

Reply:
Don't think perf will be noticeable given the max amount of 50 modules. Worst case being 50 old and 50 new. I can move into a single clause, was just hard to read for me.